tests/test_sync.py

```diff
@@ -437,6 +437,64 @@ def test_sync_dry_run_would_uninstall(echo, from_line):
     echo.assert_has_calls(expected_calls, any_order=True)
 
 
+@mock.patch("piptools.sync.click.confirm")
+@mock.patch("piptools.sync.click.echo")
+def test_sync_ask_declined_would_install(echo, confirm, from_line):
+    """
+    Sync with --ask option does a dry run if the user declines
+    """
+    confirm.return_value = False
+
+    to_install = {from_line("django==1.8"), from_line("click==4.0")}
+
+    sync(to_install, set(), ask=True)
+
+    expected_calls = [
+        mock.call("Would install:"),
+        mock.call("  django==1.8"),
+        mock.call("  click==4.0"),
+    ]
+    echo.assert_has_calls(expected_calls, any_order=True)
+
+    confirm.assert_called_once_with("Would you like to proceed with these changes?")
+
+
+@mock.patch("piptools.sync.click.confirm")
+@mock.patch("piptools.sync.click.echo")
+def test_sync_ask_declined_would_uninstall(echo, confirm, from_line):
+    """
+    Sync with --ask option does a dry run if the user declines
+    """
+    confirm.return_value = False
+
+    to_uninstall = {from_line("django==1.8"), from_line("click==4.0")}
+
+    sync(set(), to_uninstall, ask=True)
+
+    expected_calls = [
+        mock.call("Would uninstall:"),
+        mock.call("  django==1.8"),
+        mock.call("  click==4.0"),
+    ]
+    echo.assert_has_calls(expected_calls, any_order=True)
+
+    confirm.assert_called_once_with("Would you like to proceed with these changes?")
+
+
+@mock.patch("piptools.sync.click.confirm")
+@mock.patch("piptools.sync.check_call")
+def test_sync_ask_accepted(check_call, confirm, from_line):
+    """
+    pip should be called as normal when the user confirms
+    """
+    confirm.return_value = True
+
+    sync({from_line("django==1.8")}, {from_line("click==4.0")}, ask=True)
+    assert check_call.call_count == 2
+
+    confirm.assert_called_once_with("Would you like to proceed with these changes?")
+
+
```

Comment:
These tests look great! 👍 Could you also add at least one functional test to the `tests/test_cli_sync.py` module?

Reply:
Done!